src/types/store.zig

```diff
@@ -109,28 +116,54 @@ pub const Store = struct {
         return self.freshFromContent(Content{ .flex_var = null });
     }
 
-    /// Create a new variable with the given descriptor
-    pub fn register(self: *Self, desc: Desc) Var {
-        const desc_idx = self.descs.insert(self.gpa, desc);
-        const slot_idx = self.slots.insert(self.gpa, .{ .root = desc_idx });
-        return Self.slotIdxToVar(slot_idx);
+    /// Create a new unbound, flexible type variable without a name at the given idx
+    ///
+    /// This function may allocate if the provided index is greater than the
+    /// current capacities of descs or slots.
+    ///
+    /// Used in canonicalization when creating type slots
+    pub fn freshAt(self: *Self, target_idx: usize) Allocator.Error!Var {
```

Comment:
I think we should consistently use `u32` for the idx values to operate the same for both 32/64bit systems.